include/micm/process/process_set.hpp

```diff
@@ -74,6 +78,15 @@ namespace micm
         const MatrixPolicy<double>& rate_constants,
         const MatrixPolicy<double>& state_variables,
         SparseMatrixPolicy<double>& jacobian) const;
+    
+    #ifdef USE_CUDA
+    template<template<class> class MatrixPolicy>
+    void CudaAddForcingTerms(
```

Comment:
I think I would prefer to have a `CudaProcessSet` class that extends the `ProcessSet` class and overrides the `AddForcingTerms()` function. Having functions called `Cuda...` is going to lead to a lot of:
```
if (CUDA) {
  do_something
} else {
  do_something_else
}
```
@K20shores - what do you think?

Reply:
Thanks @mattldawson for pointing this out. This issue may get severer when we have more and more GPU offload code (openacc, openmp target offload, kokkos, etc). Glad  that there is a way to override the same function for different CPU/GPU code and I am interested to see how it looks like.

Reply:
A PR addressing this has been merged